CONTRIBUTING.md

```diff
@@ -17,33 +17,92 @@ version 14 on ubuntu:
 sudo apt-get install postgres-server-dev-14
 ```
 
+## Installing Postgres
+
+If you already have Postgres instaleld locally, you can skip to [Install PGMQ to Postgres](#install-pgmq-to-postgres).
+
+If you need to install Postgres or want to set up a new environment for PGMQ development, [pgenv](https://github.com/theory/pgenv/) is a command line utility that makes it very easy to install and manage multiple versions of Postgres.
+ Follow the [installation instructions](https://github.com/theory/pgenv/?tab=readme-ov-file#installation) to install it.
+ If you are on MacOS, you may need link `brew link icu4c --force` in order to successfully build Postgres.
+
+Install Postgres 16.3
+
+```bash
+pgenv build 16.3
+pgenv use 16.3
+```
+
+Connect to Postgres:
+
+```bash
+psql -U postgres
+```
+
+Fresh install will have not have PGMQ installed.
+
+```psql
+postgres=# \dx
+                 List of installed extensions
+  Name   | Version |   Schema   |         Description          
+---------+---------+------------+------------------------------
+ plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
+(1 row)
+```
+
+## Installing PGMQ
+
 Clone the repo and change into the directory.
 
 ```bash
 git clone https://github.com/tembo-io/pgmq.git
 cd pgmq
 ```
 
-### Install to a pre-existing Postgres
+### Install PGMQ to Postgres
+
+This will install the extension to the Postgres using the `pg_config` that is currently on your `PATH`. If you have multiple versions of Postgres installed, make sure you are using the correct installation by running `make install PG_CONFIG=/path/to/pg_config`.
 
 ```bash
 make
 make install
 ```
 
-Finally, you can create the extension and get started with the example in the [README.md](README.md).
+Finally, you can create the extension and get started with the example in the [README.md](README.md#sql-examples).
 
 ```psql
 CREATE EXTENSION pgmq cascade;
 ```
 
+### Installing pg_partman (optional)
+
+If you are working with partitioned queues, you will need to install `pg_partman`.
+
+```bash
+make install-pg-partman
+```
+
```

Comment:
I think this step should be before the "Install PGMQ to Postgres" one, as `cascade` will attempt to install both extensions

Reply:
`CASCADE` no longer installs pg_partman since it is no longer declared as a hard dependency in `pgmq.control`.

Reply:
pg_partman was dropped as a hard dependency in https://github.com/tembo-io/pgmq/pull/18

Reply:
```
postgres=# \dx
                 List of installed extensions
  Name   | Version |   Schema   |         Description          
---------+---------+------------+------------------------------
 plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
(1 row)

postgres=# create extension pgmq cascade;
CREATE EXTENSION
postgres=# \dx
                                     List of installed extensions
  Name   | Version |   Schema   |                             Description                             
---------+---------+------------+---------------------------------------------------------------------
 pgmq    | 1.3.3   | pgmq       | A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.
 plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
(2 rows)
```

Reply:
Ohh, I see! But then you should ask them to CREATE EXTENSION pg_partman too, right?

Reply:
Ah, I see you added